plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -3519,7 +3519,8 @@ private OptionalLong executeDelete(ConnectorSession session, ConnectorTableHandl
     private List<AddFileEntry> getAddFileEntriesMatchingEnforcedPartitionConstraint(ConnectorSession session, DeltaLakeTableHandle tableHandle)
     {
         TableSnapshot tableSnapshot = getSnapshot(session, tableHandle);
-        List<AddFileEntry> validDataFiles = transactionLogAccess.getActiveFiles(tableSnapshot, tableHandle.getMetadataEntry(), tableHandle.getProtocolEntry(), session);
+        // TODO Consider passing DeltaLakeTableHandle.getEnforcedPartitionConstraint to getActiveFiles method
```

Comment:
https://github.com/trinodb/trino/issues/19586

Reply:
It would be nice to have it for writing a test similar to `io.trino.plugin.deltalake.transactionlog.checkpoint.TestCheckpointEntryIterator#testSkipRemoveEntries` for pruning `add` entries.

Reply:
Results of your work look very promising

Tested with https://github.com/trinodb/trino/pull/19662

```
    @Test
    public void testSkipAddEntriesThroughPartitionPruning()
            throws IOException
    {
        MetadataEntry metadataEntry = new MetadataEntry(
                "metadataId",
                "metadataName",
                "metadataDescription",
                new MetadataEntry.Format(
                        "metadataFormatProvider",
                        ImmutableMap.of()),
                "{\"type\":\"struct\",\"fields\":" +
                        "[{\"name\":\"ts\",\"type\":\"timestamp\",\"nullable\":true,\"metadata\":{}}," +
                        "{\"name\":\"part_key\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}}]}",
                ImmutableList.of("part_key"),
                ImmutableMap.of(),
                1000);
        ProtocolEntry protocolEntry = new ProtocolEntry(1, 1, Optional.empty(), Optional.empty());
        int numAddEntries = 100;
        Set<AddFileEntry> addFileEntries = IntStream.rangeClosed(1, numAddEntries).mapToObj(x ->
                new AddFileEntry(
                "addFilePath",
                ImmutableMap.of("part_key", Integer.toString(x)),
                1000,
                1001,
                true,
                Optional.of("{" +
                        "\"numRecords\":20," +
                        "\"minValues\":{" +
                        "\"ts\":\"1990-10-31T01:00:00.000Z\"" +
                        "}," +
                        "\"maxValues\":{" +
                        "\"ts\":\"1990-10-31T02:00:00.000Z\"" +
                        "}," +
                        "\"nullCount\":{" +
                        "\"ts\":1" +
                        "}}"),
                Optional.empty(),
                ImmutableMap.of(),
                Optional.empty()))
                .collect(toImmutableSet());;

        CheckpointEntries entries = new CheckpointEntries(
                metadataEntry,
                protocolEntry,
                ImmutableSet.of(),
                addFileEntries,
                ImmutableSet.of());

        CheckpointWriter writer = new CheckpointWriter(
                TESTING_TYPE_MANAGER,
                checkpointSchemaManager,
                "test",
                ParquetWriterOptions.builder() // approximately 2 rows per row group
                        .setMaxBlockSize(DataSize.ofBytes(64L))
                        .setMaxPageSize(DataSize.ofBytes(64L))
                        .build());

        File targetFile = File.createTempFile("testSkipAddEntries-", ".checkpoint.parquet");
        targetFile.deleteOnExit();

        String targetPath = "file://" + targetFile.getAbsolutePath();
        targetFile.delete(); // file must not exist when writer is called
        writer.write(SESSION, entries, createOutputFile(targetPath));

        CheckpointEntryIterator metadataAndProtocolEntryIterator =
                createCheckpointEntryIterator(URI.create(targetPath), ImmutableSet.of(METADATA, PROTOCOL), Optional.empty(), Optional.empty(), TupleDomain.all());

        DeltaLakeColumnHandle partitionKeyField = new DeltaLakeColumnHandle(
                "part_key",
                INTEGER,
                OptionalInt.empty(),
                "part_key",
                INTEGER,
                REGULAR,
                Optional.empty());

        CheckpointEntryIterator addEntryIterator = createCheckpointEntryIterator(
                URI.create(targetPath),
                ImmutableSet.of(ADD),
                Optional.of(metadataEntry),
                Optional.of(protocolEntry),
                TupleDomain.all());

        CheckpointEntryIterator addEntryIteratorWithPartitionFilter = createCheckpointEntryIterator(
                URI.create(targetPath),
                ImmutableSet.of(ADD),
                Optional.of(metadataEntry),
                Optional.of(protocolEntry),
                TupleDomain.withColumnDomains(ImmutableMap.of(partitionKeyField, singleValue(INTEGER, 10L))));

        assertThat(Iterators.size(metadataAndProtocolEntryIterator)).isEqualTo(2);
        assertThat(Iterators.size(addEntryIterator)).isEqualTo(numAddEntries);
        assertThat(Iterators.size(addEntryIteratorWithPartitionFilter)).isEqualTo(1);

        assertThat(metadataAndProtocolEntryIterator.getCompletedPositions().orElseThrow()).isEqualTo(3L);
        assertThat(addEntryIterator.getCompletedPositions().orElseThrow()).isEqualTo(101L);
        assertThat(addEntryIteratorWithPartitionFilter.getCompletedPositions().orElseThrow()).isEqualTo(1L);
    }
```

Reply:
Tried as well the partition pruning on a multipart checkpoint suming around 300MB (composed out of 25 parts) and got the following very promising results:

```
ADD
number of add entries retrieved:1.235.155
completed positions: Optional[1235157]
completed bytes: Optional[323227977]
Elapsed Time in milliseconds: 15084
ADD PRUNING
number of add entries retrieved: 18.578
completed positions: Optional[49290]
completed bytes: Optional[14099674]
Elapsed Time in milliseconds: 919
```